guides/gitconfig.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Managing Git Configuration
+description: Learn how to manage your Git configuration in Coder.
+---
+
+This guide will show you how to manage Git configuration through the system,
+global, and repo levels.
+
+## How to use the System and Global configurations
+
+For customers seeking organizational (system-level) `.gitconfig` defaults, we recommend
+adding a default configuration into the `/etc/gitconfig` file of each image associated
+to the organization:
+
+```bash
+[url "git@github.com:"]
+    insteadOf = https://github.com/
+```
```

Comment:
This is not a `bash` code fence. This is `gitconfig`. Furthermore, the example should prefaced that this is an example `.gitconfig` file and the example should explain what it's doing.

Reply:
Do we want to provide examples? There are a lot of guides out there on `gitconfig` and while we can provide one too, it's not necessarily our area of expertise as a company, right? Like the version control you use doesn't matter with Coder. There's a tighter integration with 3 major git cloud platforms, but you could use SVN or Perforce too.

Open to thoughts on this, and I like the idea of a doc, but I'm curious how much we should maintain versus acknowledging that `git` knowledge is outside the scope of our product in a general sense.

Reply:
Agree as well - will change to something like:

> Below is an example of `.gitconfig` contents that could be applied to `/etc/gitconfig`. Here we are making `ssh` the System-level default when engaging with Git providers.

```gitconfig
[url "git@github.com:"]
    insteadOf = https://github.com/
```

Let me know if this works or if I'm off-base here.

Reply:
> Do we want to provide examples? There are a lot of guides out there on `gitconfig` and while we can provide one too, it's not necessarily our area of expertise as a company, right? Like the version control you use doesn't matter with Coder. There's a tighter integration with 3 major git cloud platforms, but you could use SVN or Perforce too.
> 
> Open to thoughts on this, and I like the idea of a doc, but I'm curious how much we should maintain versus acknowledging that `git` knowledge is outside the scope of our product in a general sense.

Understand your point here, but the purpose is to provide context on `.gitconfig` as it relates to Coder & how users can manage their git configs thought the various levels (system, global, etc.)

Reply:
Right, understandable. I don't think it's a bad thing to provide info on, but `--system`, `--local`, `--global` aren't unique to Coder was my point.

Reply:
Given that we inject a default `.gitconfig` and that we adjusted build order based off of end-user feedback, I think an example could be useful but I'm also not against scrapping the example.

I'll leave that decision to y'all.

My only point is that if we keep the example in, we change the code fences, mention it's example contents of a `.gitconfig` for using `ssh` as default within an example Coder environment. We could even say we ourselves use this setting if we feel it's beneficial to mention so.

OTherwise, ya, I'll defer to @IAmHughes @khorne3  and @ericpaulsen

Reply:
> My only point is that if we keep the example in, we change the code fences, mention it's example contents of a .gitconfig for using ssh as default within an example Coder environment. We could even say we ourselves use this setting if we feel it's beneficial to mention so.

Totally on board with that approach!